sdk/servicebus/service-bus/test/stress/stress_lockRenewal.ts

```diff
@@ -0,0 +1,107 @@
+/*
+Test Scenario summary:
+- Creates a single sender and a single receiver on a queue.
+- Receives a message and holds onto it for the duration of test. AutoLockRenewal is enabled and set to the test duration.
+
+The test assumes no other process is working with the queues defined in here,
+but the queues must be empty and use default configurations before running the test.
+
+For running this test, connection string of the Service Bus namespace and queue name
+must be supplied.
+*/
+
+import {
+  ServiceBusClient,
+  SendableMessageInfo,
+  OnMessage,
+  OnError,
+  delay,
+  ReceiveMode,
+  ServiceBusMessage
+} from "../../src";
+
+const connectionString = "";
+const queueName = "";
+
+const testDurationInMilliseconds = 60000 * 10;
+
+let snapshotIntervalID: any;
+
+let receivedMessage: ServiceBusMessage;
+
+async function main(): Promise<void> {
+  snapshotIntervalID = setInterval(snapshot, 60000); // Every 60 seconds
+  await sendMessage();
+  await receiveMessage();
+}
+
+async function sendMessage(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+  try {
+    const sender = client.createSender();
+
+    const message: SendableMessageInfo = {
+      messageId: "test",
+      body: "test",
+      label: `test`
+    };
+
+    await sender.send(message);
+  } finally {
+    await client.close();
+    await ns.close();
+  }
+}
+
+async function receiveMessage(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+
+  try {
+    const receiver = client.createReceiver(ReceiveMode.peekLock);
+    const onMessageHandler: OnMessage = async (brokeredMessage) => {
+      receivedMessage = brokeredMessage;
+      if (receivedMessage.messageId !== "test") {
+        throw new Error("Message is corrupt or unexpected");
+      }
+      console.log("Received message: ", receivedMessage.messageId);
+
+      await delay(testDurationInMilliseconds);
```

Comment:
One thing I would do here is make it so that we go into a loop and every x seconds we log something. that way we can tell that the customer code continues to run.

Reply:
Yes, the snapshot() function does just this. It gets switched off when program exits.

Reply:
But the snapshot function doesn’t run inside this message handler, right? The snapshot function is started outside of this message handler.

I want to make sure that the user code continues to run while we renew the lock.

Reply:
> I want to make sure that the user code continues to run while we renew the lock.

Yes, this is currently the case because we are awaiting inside the user code. Should the user code exit, we'll know by looking at the logs.

What kind of user code are you looking to test? Blocking, non-blocking?

Reply:
What I was thinking is to have something like this:

```typescript
let elapsedTime = 0;
const interval = 10 * 1000; // 10 seconds
while (elapsedTime < testDurationInMiliseconds)
{
   // simulate the user making an async call that takes time. 
   await delay(interval); 
   elapsedTime += interval;
  
   // log how long we've executed.
   console.log(`still executing after {elapsedTime}`);
}
```

Reply:
Here `await delay(testDurationInMilliseconds);` represents the user code that @AlexGhiondea is talking about. Unfortunately it does nothing but wait. 

If we want this part to print out things, then instead of the single delay we can do what @AlexGhiondea  is suggesting above.

The only advantage this has is that whatever the lock renewal is doing is not blocking the user's code.

Reply:
Got it, please see - https://github.com/Azure/azure-sdk-for-js/issues/2252#issuecomment-483846785
There's an issue that's surfaced with and without these changes.

Reply:
Let's update the code to do some work other than a single delay like I suggested above.